adapters/mqtt-vertx/src/main/java/org/eclipse/hono/adapter/mqtt/impl/VertxBasedMqttProtocolAdapter.java

```diff
@@ -49,6 +52,23 @@ protected String getTypeName() {
         return Constants.PROTOCOL_ADAPTER_TYPE_MQTT;
     }
 
+    private WebClient getWebClient() {
+        if (webClient == null) {
+            webClient = WebClient.create(vertx);
+        }
+        return webClient;
+    }
+
+    /**
+     * Sets the vertx webClient to use when mapping payload.
+     *
+     * @param webClient the webClient to set
+     */
+    public void setWebClient(final WebClient webClient) {
```

Comment:
this should be autowired and injected via Spring FMPOV

Reply:
In fact, I believe that it makes a lot of sense to create a dedicated interface for message mapping that hides away the specifics of how the mapper is invoked (HTTP in this case) and which also encapsulates the configuration of the supported mapper endpoints. WDYT?

Reply:
I agree. I've created another class for this. However, I can't seem to autowire the `WebClient` in it. Can you point me to where I should configure this?